pkg/controller/core/localqueue_controller.go

```diff
@@ -270,26 +270,32 @@ func (r *LocalQueueReconciler) initializeAdmissionFsStatus(ctx context.Context,
 	return false
 }
 
-func (r *LocalQueueReconciler) reconcileConsumedUsage(ctx context.Context, lq *kueue.LocalQueue, cqName kueue.ClusterQueueReference) error {
-	halfLifeTime := r.admissionFSConfig.UsageHalfLifeTime.Seconds()
+func (r *LocalQueueReconciler) reconcileConsumedUsage(ctx context.Context, lq *kueue.LocalQueue) error {
+	penalty := r.cache.GetAndClearEntryPenalty(lq)
```

Comment:
If at some point the `reconcileConsumedUsage` fails and it doesn't manage to make changes to the LQ's status this penalty will be dropped permanently. I believe we should have some mechanism to restore in case of failures

Reply:
We should take the mutex until the function finishes so it's synchronized correctly with the scheduler and there are no race conditions. Also please move getting the value right before using it

Reply:
The scenario I'm worried about is as following:

1. LQ controller pops the penalty
2. Scheduler starts its loop, and it's done without the penalty as it's removed at this moment
3. LQ controller fails for some reason and then put the penalty back

I'd like to have a mechanism that ensures the two components are synchronized correctly. I'm open to discuss ideas and here's my proposal:

Popping function should take the mutex but it should release it. Instead, it should return the function that releases the mutex, and after LQ controller/scheduler finish their operations they call the function.